### ISSUE TITLE
"panic: assignment to entry in nil map" on SetAuthentication 

### ISSUE THREAD

**Issue by CooperLi** [containers/image#972]
**Description**

Modify `auth.json` fields leads to `panic: assignment to entry in nil map`

**Steps to reproduce the issue:**

1. success login image registry

2. change `auth.json` fileds

    ```
    $ cat auth.json           
    {
            "auths": {
                    "mydockerhub.io": {
                            "auth": "aGVsbG86d29ybGQK"
                    }
    }
    
    # after modified
    $ diff auth.json auth.json.bk
    2c2
    <       "auths": {
    ---
    >       "buths": {
    
    ```

3. continue login with same registry

4. panic

**Describe the results you received:**

```bash
panic: assignment to entry in nil map

goroutine 168 [running]:
github.com/containers/image/v5/pkg/docker/config.SetAuthentication.func1(0xc0004bc050, 0x1a, 0x12e6d00, 0x0)
	/home/GOWORK/src/cooper/cooper/vendor/github.com/containers/image/v5/pkg/docker/config/config.go:71 +0x1b2
github.com/containers/image/v5/pkg/docker/config.modifyJSON(0xc00046a8c0, 0xc000327970, 0xc37416, 0x0)
	/home/GOWORK/src/cooper/cooper/vendor/github.com/containers/image/v5/pkg/docker/config/config.go:276 +0x198
github.com/containers/image/v5/pkg/docker/config.SetAuthentication(0xc00046a8c0, 0xc000474b80, 0x18, 0xc0002e2948, 0x7, 0xc0002e2990, 0xa, 0xc000474b80, 0x18)
	/home/GOWORK/src/cooper/cooper/vendor/github.com/containers/image/v5/pkg/docker/config/config.go:52 +0xa8
```

**Describe the results you expected:**
no panic, login success with right auth info; `auth.json` will be recover to the right form or error poped out

**Additional information you deem important (e.g. issue happens only occasionally):**

- issue will `always happened` as long as the fileds in auth.json was modified

- go.mod:  `github.com/containers/image/v5 v5.4.4`

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Virtual Machine (CentOS7.5)

**Comment by CooperLi** [comment]
It seems start with: 
`vendor/github.com/containers/image/v5/pkg/docker/config/config.go:228`

```go
func readJSONFile(path string, legacyFormat bool) (dockerConfigFile, error) {
	var auths dockerConfigFile
 	/* should set map here?
 	   no matter the path is exist or not, we need to ensure the auths.AuthConfigs is not nil
 	*/
	raw, err := ioutil.ReadFile(path)
	if err != nil {
		if os.IsNotExist(err) {
			auths.AuthConfigs = map[string]dockerAuthConfig{}        // <== too late to set map
			return auths, nil
		}
		return dockerConfigFile{}, err
	}
```

**Comment by mtrmac** [comment]
Thanks. for your report. I think this was recently fixed in #942 , see https://github.com/containers/image/pull/973#pullrequestreview-440330405 .
